### DISCUSSION TITLE
How to declare a column using `gen_random_uuid` in postgres?

### DISCUSSION THREAD

**Discussion by leonardbinet** [mikro-orm/mikro-orm#4403] [Q&A] [answered]
Given the following SQL 
```
ALTER TABLE "user" ADD "uuid" uuid NOT NULL DEFAULT uuid_generate_v4()
CREATE UNIQUE INDEX "IDX_unique_user_uuid" ON "user" ("uuid") 
```

How should I declare it in mikro-orm?

Based on https://mikro-orm.io/docs/defining-entities#using-uuid-as-primary-key-sql-drivers, it seems:
```
@Property({ index: true })
readonly uuid = v4()
```

it seems functional, but doesn't really match the intended behavior since uuid is generated in application, not in postgres using `uuid_generate_v4`.

https://www.postgresql.org/docs/current/functions-uuid.html

**Comment by B4nan** [comment, marked as answer]
```ts
@Property({ type: 'uuid', defaultRaw: 'uuid_generate_v4()' })
readonly uuid: string = v4()
```

**Reply by B4nan** [reply]
Sounds like you didn't really read the docs, as this is on the page you linked yourself :]

<img width="1024" alt="image" src="https://github.com/mikro-orm/mikro-orm/assets/615580/8a47aa3e-9cbb-41c8-ad5a-f53261b24b14">

**Reply by leonardbinet** [reply]
I've seen it (the title using postgresql uuid-ossp module, with info saying that it requires the uuid-ossp extension) but didn't read the code, I think the title is misleading.

<img width="755" alt="Screenshot 2023-05-30 at 13 58 19" src="https://github.com/mikro-orm/mikro-orm/assets/1590371/12332e63-b184-4cc6-8fb1-e0245f5bd66b">

As the [postgres documentation](https://www.postgresql.org/docs/current/uuid-ossp.html) for this extensions states it:
> The uuid-ossp module provides functions to generate universally unique identifiers (UUIDs) using one of several standard algorithms. There are also functions to produce certain special UUID constants. This module is only necessary for special requirements beyond what is available in core PostgreSQL. See [Section 9.14](https://www.postgresql.org/docs/current/functions-uuid.html) for built-in ways to generate UUIDs.


`uuuis-ossp` module is different from the builtin `uuid_generate_v4`
